python_modules/dagster/dagster/_core/definitions/declarative_automation/operands/operands.py

```diff
@@ -148,8 +152,39 @@ def name(self) -> str:
         return "executed_with_root_target"
 
     async def compute_subset(self, context: AutomationContext) -> EntitySubset:
-        return await context.asset_graph_view.compute_latest_run_executed_with_subset(
-            from_subset=context.candidate_subset, target=context.root_context.key
+        def _filter_fn(run_record: "RunRecord") -> bool:
+            asset_selection = run_record.dagster_run.asset_selection or set()
+            check_selection = run_record.dagster_run.asset_check_selection or set()
+            return context.root_context.key in (asset_selection | check_selection)
+
+        return await context.asset_graph_view.compute_latest_run_matches_subset(
+            from_subset=context.candidate_subset, filter_fn=_filter_fn
+        )
+
+
+@whitelist_for_serdes
+@record
+class LatestRunExecutedWithTagsCondition(SubsetAutomationCondition):
+    tag_keys: Optional[Set[str]] = None
+    tag_values: Optional[Mapping[str, str]] = None
```

Comment:
just naming wise, when i read this i initially assumed that if you wanted to do filtering on both keys and values you would put the keys in tag_keys and the values in tag_values (even though that doesn't really make sense) - should `tag_values` be `tag_keys_and_values`? Or maybe this should even be two separate conditions?

Reply:
I see where you're coming from, I'm leaning towards keeping the current name as I think the example looks (subjectively) a bit better with just `tag_values`, i.e.:

```python
executed_via_condition = dg.AutomationCondition.executed_with_tags(
    tag_values={"dagster/from_automation_condition": "true"}
)
```

which feels a bit more natural once you see it written out (i.e. you're specifying tag values for multiple keys at once). 

but if you have reservations about this I don't have a strong objection to your suggestion other than verbosity